litmus-portal/graphql-server/graph/schema.resolvers.go

```diff
@@ -123,7 +122,13 @@ func (r *mutationResolver) SaveMyHub(ctx context.Context, myhubInput model.Creat
 	return myhub.SaveMyHub(ctx, myhubInput, projectID)
 }
 
-func (r *mutationResolver) SyncHub(ctx context.Context, id string) ([]*model.MyHubStatus, error) {
+func (r *mutationResolver) SyncHub(ctx context.Context, id string, projectID string) ([]*model.MyHubStatus, error) {
+	err := authorization.ValidateRole(ctx, projectID,
+		authorization.MutationRbacRules[authorization.UpdateChaosWorkflow],
+		model.InvitationAccepted.String())
+	if err != nil {
+		return nil, err
+	}
```

Comment:
Will not work creates attack vector. I can send a random project id and an unrelated hub id, the project id should be derived from the hub id

Reply:
A simple solution would be to query the hub with the hub id + project id, this would mean if the user sends unrelated hub and project ids db will not return the hub data.